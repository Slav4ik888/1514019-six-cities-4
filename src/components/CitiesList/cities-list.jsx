import React from 'react';
import {Link} from 'react-router-dom';
import {AppRoute} from '../../utils/const.js';
import PropTypes from 'prop-types';
import {cities} from '../../utils/const.js';

const CitiesList = (props) => {
  const {activeCity, onItemClick} = props;

  const isActive = `tabs__item--active`;

  return (
    <>
      <h1 className="visually-hidden">Cities</h1>
      <div className="tabs">
        <section className="locations container">
          <ul className="locations__list tabs__list">

            {cities.map((item, i) => {
              return (
                <li key={item} className="locations__item"
                  onClick={() => (onItemClick(i))}
                >
                  <Link className={`locations__item-link tabs__item
                  ${(activeCity === i) && isActive}`}
                  to={AppRoute.ROOT}>
                    <span>{item}</span>
                  </Link>
                </li>
              );
            })}

          </ul>
        </section>
      </div>
    </>
  );
};

CitiesList.propTypes = {
  activeCity: PropTypes.number.isRequired,
  onItemClick: PropTypes.func.isRequired,
};


export default CitiesList;
