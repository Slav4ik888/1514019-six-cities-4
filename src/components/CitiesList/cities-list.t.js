import React from 'react';
import renderer from 'react-test-renderer';
import CitiesList from './cities-list.jsx';


describe(`Snapshot of <CitiesList>`, () => {
  it(`Render <CitiesList>`, () => {
    const tree = renderer
      .create(
          <CitiesList
            activeCity={0}
            onItemClick={() =>{}}
          />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
