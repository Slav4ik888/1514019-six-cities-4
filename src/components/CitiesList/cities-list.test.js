import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import CitiesList from './cities-list.jsx';


describe(`Snapshot <CitiesList/>`, () => {
  it(`Render <CitiesList/>`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <CitiesList
              activeCity={0}
              onItemClick={() =>{}}
            />
          </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

// npm run test.jest -- -u card-list.test.js
// npm test card-list.test.js
