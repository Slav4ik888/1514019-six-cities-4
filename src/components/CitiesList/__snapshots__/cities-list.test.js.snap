// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of <CitiesList> Render <CitiesList> 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>,
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            className="locations__item-link tabs__item
                  tabs__item--active"
            href="#"
          >
            <span>
              Paris
            </span>
          </a>
        </li>
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            className="locations__item-link tabs__item
                  false"
            href="#"
          >
            <span>
              Cologne
            </span>
          </a>
        </li>
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            className="locations__item-link tabs__item
                  false"
            href="#"
          >
            <span>
              Brussels
            </span>
          </a>
        </li>
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            className="locations__item-link tabs__item
                  false"
            href="#"
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            className="locations__item-link tabs__item
                  false"
            href="#"
          >
            <span>
              Hamburg
            </span>
          </a>
        </li>
        <li
          className="locations__item"
          onClick={[Function]}
        >
          <a
            className="locations__item-link tabs__item
                  false"
            href="#"
          >
            <span>
              Dusseldorf
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>,
]
`;
